pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -130,17 +130,21 @@ func resolveResultRef(pipelineState PipelineRunState, resultRef *v1beta1.ResultR
 	var resultValue v1beta1.ResultValue
 	var err error
 	if referencedPipelineTask.IsCustomTask() {
-		runName = referencedPipelineTask.RunObject.GetObjectMeta().GetName()
-		runValue, err = findRunResultForParam(referencedPipelineTask.RunObject, resultRef)
-		resultValue = *v1beta1.NewStructuredValues(runValue)
-		if err != nil {
-			return nil, resultRef.PipelineTask, err
+		for _, runObject := range referencedPipelineTask.RunObjects {
```

Comment:
I'm not sure what I said earlier-- I think I was confused about using runObjects[0] since matrix doesn't support full array results, but at first glance this looks like it works for matrix even though it doesn't. I think it would make more sense to check if len = 1, return an error if not, and add a comment explaining why.

Reply:
Okay, updated with error handling for len != 1. I wasn't sure how to test this previously, since right now it is impossible to have plural runObjects/taskRuns created since validation prevents a Matrix from producing results but I think I figured out a way to test.